src/lte/app_link_handler.c

```diff
@@ -0,0 +1,78 @@
+#include <zephyr/logging/log.h>
```

Comment:
The only real change here is getting rid of noise in logs.

Ultimately I would like to have the functionality we're after [added to the common library link handler](https://github.com/golioth/golioth-firmware-sdk/blob/main/examples/zephyr/common/nrf91_lte_monitor.c). But for now I think moving this code to a file helps make main.c easier to follow.

This is a pretty large change, and maybe should have been a PR outside of the migration effort. LMK if you want this separated from the rest.

Reply:
This commit was approved earlier today so it should be available soon: https://github.com/golioth/golioth-firmware-sdk/pull/334

Reply:
> LMK if you want this separated from the rest.

No strong opinions either way, I'm fine if you want to keep it included.

> Ultimately I would like to have the functionality we're after added to the common library link handler.

If/when we do that, it would be nice to provide a `register_lte_on_disconnect_cb` too.

Reply:
Good idea. Maybe the same callback but the event gets passed into the callback? This is the pattern currently used by the Golioth connection event callback.

Reply:
Yeah, good advice. [When originally added](https://github.com/golioth/reference-design-template/commit/df5d4e6f07b72a520ce7c9f3910c0dccd9653d44) we didn't have control of when the LTE connection was started. But this code has changed since then and I think we can use that now. I'll work on moving over to it.

This saves a log of complexity, thank you!

Reply:
@cdwilson Great suggestion, this code is now much simpler!!

I will work on getting a setting into the upstream SDK for quieting the LTE logs, but this repo is ready for review.